src/endianity.rs

```diff
@@ -252,6 +255,17 @@ impl<'input, Endian> Into<&'input [u8]> for EndianBuf<'input, Endian>
     }
 }
 
+impl<'input, Endian> Read for EndianBuf<'input, Endian>
+    where Endian: Endianity
+{
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
```

Comment:
>Takes a performance hit though, so may want to inline it later.

Do you mean _manually_ inlining this code? Did you try adding `#[inline]` here? Any difference?